packages/relay/src/lib/db/repositories/hbarLimiter/ipAddressHbarSpendingPlanRepository.ts

```diff
@@ -102,6 +102,36 @@ export class IPAddressHbarSpendingPlanRepository {
     }
   }
 
+  /**
+   * Deletes all IP address spending plans from the cache.
+   *
+   * @param {RequestDetails} requestDetails - The request details for logging and tracking.
+   * @returns {Promise<void>} - A promise that resolves when all IP address spending plans are deleted.
+   */
+  async deleteAll(requestDetails: RequestDetails): Promise<void> {
+    const key = this.getKey('*');
+    const keys = await this.cache.keys(key, 'deleteAll', requestDetails);
+    for (const key of keys) {
+      await this.cache.delete(key, 'deleteAll', requestDetails);
+    }
+    this.logger.trace(`${requestDetails.formattedRequestId} Deleted all IP address spending plans`);
+  }
+
+  /**
+   * Gets all IP address spending plans from the cache.
+   *
+   * @param {RequestDetails} requestDetails - The request details for logging and tracking.
+   * @returns {Promise<IPAddressHbarSpendingPlan[]>} - A promise that resolves with an array of IP address spending plans.
+   */
+  async getAllPlans(requestDetails: RequestDetails): Promise<IPAddressHbarSpendingPlan[]> {
+    const pattern = `${this.collectionKey}:*`;
+    const keys = await this.cache.keys(pattern, 'getAllPlans', requestDetails);
+    const plans = await Promise.all(
+      keys.map((key) => this.cache.getAsync<IIPAddressHbarSpendingPlan>(key, 'getAllPlans', requestDetails)),
```

Comment:
Let's try to use the `name` property of this particular method instead of a string. This way if the method name is ever updated, we won't have to worry about updating the string.

```suggestion
    keys.map((key) => this.cache.getAsync<IIPAddressHbarSpendingPlan>(key, this.getAllPlans.name, requestDetails)),
```